src/ReaderScreen/index.jsx

```diff
@@ -37,69 +34,98 @@ const Reader = ({ navigation, route }) => {
   const isVishraam = useSelector((state) => state.isVishraam);
   const vishraamOption = useSelector((state) => state.vishraamOption);
   const savePosition = useSelector((state) => state.savePosition);
-  const isHeaderFooter = useSelector((state) => state.isHeaderFooter);
+  const theme = useSelector((state) => state.theme);
+
+  const webViewRef = useRef(null);
+  const { webView } = styles;
+  const { title, id } = route.params.params;
+  const [error, setError] = useState(null);
+  const [isHeader, toggleHeader] = useState(false);
+  const [viewLoaded, toggleViewLoaded] = useState(false);
+  const [currentPosition, setCurrentPosition] = useState(savePosition[id] || 0);
+
+  const dispatch = useDispatch();
+
+  const { shabad, isLoading, fetchShabad } = useFetchShabad(id, setError);
 
-  const { shabad, isLoading } = useFetchShabad(shabadID);
-  const [currentPosition, setCurrentPosition] = useState(savePosition[shabadID] || 0);
   const { backgroundColor, safeAreaViewBack, backViewColor } = nightColors(isNightMode);
   const { READER_STATUS_BAR_COLOR } = colors;
+
+  if (error) {
+    errorHandler(error);
+    throw error;
+  }
+
+  const updateTheme = () => {
+    const currentColorScheme = Appearance.getColorScheme();
+    if (theme === constant.Default) {
+      dispatch(actions.toggleNightMode(currentColorScheme === "dark"));
+    }
+  };
+
   useScreenAnalytics(title);
   useBookmarks(webViewRef, shabad, bookmarkPosition);
 
   useEffect(() => {
-    setShabadID(Number(id));
-  }, [id]);
-
-  useEffect(() => {
-    setCurrentPosition(savePosition[shabadID]);
-    if (Number(savePosition[shabadID]) > 0.9) {
-      setCurrentPosition(0);
+    if (savePosition && id) {
+      setCurrentPosition(savePosition[id]);
+      if (Number(savePosition[id]) > 0.9) {
+        setCurrentPosition(0);
+      }
     }
-  }, []);
+  }, [savePosition, id]);
+
   useEffect(() => {
-    if (headerRef.current && headerRef.current.toggle) {
-      headerRef.current.toggle(isHeader);
-    }
-  }, [isHeader]);
+    const subscription = AppState.addEventListener("change", (state) => {
+      if (state === "active") {
+        updateTheme();
+        fetchShabad();
+      }
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  }, [fetchShabad]);
 
   const handleBackPress = () => {
-    webViewRef.current.postMessage(JSON.stringify({ Back: true }));
-    setTimeout(() => {
-      navigation.goBack();
-    }, 100);
+    if (webViewRef) {
+      webViewRef.current.postMessage(JSON.stringify({ Back: true }));
+      // set Timeout to delay back function so that it will save the current position
+      setTimeout(() => {
+        navigation.goBack();
+      }, 100);
```

Comment:
Can we use promises to handle this instead of using a setTimeout?

Reply:
nopes veer ji we can't use promise here as they are not returning any value.

Reply:
Can you please point me to where we save the current position in code.